components/nimbus/ios/Nimbus/Operation+.swift

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+public extension Operation {
+    /// Wait for the operation to finish, or a timeout.
+    ///
+    /// The operation is cooperatively cancelled on timeout, that is to say, it checks its {isCancelled}.
+    func joinOrTimeout(timeout: TimeInterval) -> Bool {
```

Comment:
I _really_ like the re-impl of the Kotlin method here. It really helps the code flow well in the `build` method!